sklearn/gaussian_process/kernels.py

```diff
@@ -0,0 +1,475 @@
+""" Kernels for Gaussian process regression and classification.
+
+The kernels in this module allow kernel-engineering, i.e., they can be
+combined via the "+" and "*" operators. These expressions can also contain
+scalar values, which are automatically converted to a constant kernel.
+
+All kernel allow (analytic) gradient-based hyperparameter optimization.
+The space of hyperparameters can be specified by giving lower und upper
+boundaries for the value of each hyperparameter (the search space is thus
+rectangular). This can be achieved by using a pair or triple instead of a
+single float wherever a parameter value is specified. In case of a pair,
+the first value specifies the lower boundary and the second value the upper
+boundary. In case of a triple, the middle value specified the initial value
+of the parameter during hyperparameter-optimization.
```

Comment:
I am also not really satisfied. In the george package, the parameter space is always R^+, which simplifies kernel specification but lacks flexibility. We could make R^+ the default and require the pair/triple specification only if one wants to deviate from this. But then it is unclear how to specify that certain parameters should be kept fixed.